File: leetcode/1. Two Sum/solution.py
class Solution():
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        num_dict = {}
        for i in range(len(nums)):
            num_dict[nums[i]] = i
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in num_dict and num_dict[complement] != i:
                return [i, num_dict[complement]]
        return []

        

n, target = input().split()
n = int(n)
target = int(target)
nums = list(map(int, input().split()))
sol = Solution()
result = sol.twoSum(nums, target)
print(result[0], result[1])

File: leetcode/1. Two Sum/solution.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        int diff;
        vector<int> ans;
        map<int, int> mp;
        for (int i = 0; i < nums.size(); i++)
        {
            mp[nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++)
        {
            diff = target - nums[i];
            if (mp.count(diff) && mp[diff] != i)
            {
                return {i, mp[diff]};
            }
        }
        return {};
    }
};

int main()
{
    int n, target;
    cin >> n >> target;
    vector<int> nums;
    for (int i = 0; i < n; i++)
    {
        int x;
        cin >> x;
        nums.push_back(x);
    }
    Solution sol;
    vector<int> ans = sol.twoSum(nums, target);
    for (int i = 0; i < 2; i++)
    {
        cout << ans[i] << " \n"[i == 1];
    }
    return 0;
}